### PR TITLE
sqlpp11-connector-sqlite3: add 0.30 + fix hooks + cache cmake + del fPIC if shared

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **sqlpp11-connector-sqlite3/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sqlpp11-connector-sqlite3/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 sources:
   "0.29":
     url: "https://github.com/rbock/sqlpp11-connector-sqlite3/archive/0.29.tar.gz"
-    sha256: 2904ddd578adb0e122a70f73a80bae4843434d685b7a23ae3ab43471c35efc4f
+    sha256: "2904ddd578adb0e122a70f73a80bae4843434d685b7a23ae3ab43471c35efc4f"
+  "0.30":
```
Comment (madebr) [inline recipes/sqlpp11-connector-sqlite3/all/conandata.yml:5]: Hey @SpaceIm 
In my latest recipes, I'm adding the most recent versions at the top.
That makes making changes easier, imho.
What do you think?

Comment (SpaceIm) [reply recipes/sqlpp11-connector-sqlite3/all/conandata.yml:5]: I was doing the same things in the past, then I saw some automated script of @SSE4 (not sure) running and reordering conandata.yml and config.yml from older to latest. But I agree that latest first makes more sense.

Comment (madebr) [reply recipes/sqlpp11-connector-sqlite3/all/conandata.yml:5]: Maybe @Croydon 's @CroydonBot

Comment (Croydon) [reply recipes/sqlpp11-connector-sqlite3/all/conandata.yml:5]: My scripts aren't (yet) capable of adding new versions. That was probably @SSE4's scripts for adding new versions

But I don't think that it matters either way if it is ascending or descending as long as it is ordered within a recipe

**Review by uilianries — APPROVED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
The build error complains about pthread. It has to be added as system_libs.

**Comment by SpaceIm** [comment]
The shared lib complains about pthread, so it comes either from build itself, or from a system libs missing in a static dependency.

**Review by danimtb — APPROVED** [review]

File: recipes/sqlpp11-connector-sqlite3/all/conandata.yml (hunk 1 above)
Comment (SSE4) [reply recipes/sqlpp11-connector-sqlite3/all/conandata.yml:5]: yes, it was mine script. the order was automatic by `yaml` python module.

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**